transcom/mymove
# MAIN B-20652 boat shipment customer flow

## [B-20652](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20652)

INT PR: https://github.com/transcom/mymove/pull/13401

## Summary

This adds Boat Shipment UI to customer app where they can select "Move a Boat" to create a boat shipment.

- Added "Move a boat" selection card.
- Added "Question" mark icon on the selection card with more details on boat shipment.
- Added boat shipment information form.
- Added a confirmation modal that shows when pressing continue button from the 1st page of the boat shipment form.
- Added 2nd page of the boat shipment form to match current HHG form.

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Create a customer and verify that "Move a boat" card is available.
2. Click "Question mark" icon next to "Move a boat" card to view more info about boat shipment in a modal.
3. Click the boat card and continue.
4. Fill out the form and test these cases:

-  Put 2 feet each for length, width, and height -> Click "Continue" and you should see a modal with HHG information and when clicking "Continue" again from the modal, it should redirect you to a HHG create form.
- Put 22 feet for any length, width, and height -> Choose "No" for "Does the boat have a trailer". -> Click "Continue" and the modal should show "Boat Haul-Away" modal and continues to the next form when clicking continue.
- Put 22 feet for any length, width, and height -> Choose "Yes" for both "Does the boat have a trailer" and "Is the trailer roadworthy". -> Click "Continue" and you should see "Boat Tow-Away" modal and continues to next form when clicking continue.

## Screenshots
![image](https://github.com/user-attachments/assets/1438ca5f-8783-4c95-bf04-1dde76d40ce9)
![image](https://github.com/user-attachments/assets/2d6b2558-122d-4c47-87e6-b2b80e3e35b0)
![image](https://github.com/user-attachments/assets/a5bc6ec2-b4c5-40e7-8a70-d23eb09cbb61)
![image](https://github.com/user-attachments/assets/37f2a0a4-5d0a-4248-b44f-25ef57574cd2)
![image](https://github.com/user-attachments/assets/1f7c41c1-b88a-4e99-9abd-c5e5cfe3d7a0)
![image](https://github.com/user-attachments/assets/63736a42-250e-4259-934b-906b6893ba86)
![image](https://github.com/user-attachments/assets/594b0b37-4e59-42cc-b48d-9572f06a48ac)
